tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js

```diff
@@ -1,25 +1,283 @@
+const assert = require('assert')
+const util = require('util')
+
 module.exports = {
   commands: {
+    /**
+     * function clicks the edit button from public link dialog box
+     *
+     * @param linkName Name of the public link
+     * @returns {Promise<void>}
+     */
+    clickLinkEditBtn: async function (linkName) {
+      const linkRowEditButtonSelector = util.format(this.elements.publicLinkEditButton.selector, linkName)
+      const linkRowEditButton = {
+        locateStrategy: 'xpath',
+        selector: linkRowEditButtonSelector
+      }
+      await this
+        .initAjaxCounters()
+        .waitForElementVisible(linkRowEditButton)
+        .click(linkRowEditButton)
+        .waitForOutstandingAjaxCalls()
+    },
+    /**
+     * function sets up the xpath for year in expiry date of public link
+     *
+     * @param year
+     * @returns {{locateStrategy: string, selector: *}}
+     */
+    setExpiryDateYearSelectorXpath: function (year) {
+      const yearSelectorXpath = util.format(this.elements.dateTimeYearPicker.selector, year)
+      return {
+        selector: yearSelectorXpath,
+        locateStrategy: 'xpath'
+      }
+    },
+    /**
+     * function sets up the xpath for month in expiry date of public link
+     *
+     * @param month
+     * @returns {{locateStrategy: string, selector: *}}
+     */
+    setExpiryDateMonthSelectorXpath: function (month) {
+      const monthSelectorXpath = util.format(this.elements.dateTimeMonthPicker.selector, month)
+      return {
+        selector: monthSelectorXpath,
+        locateStrategy: 'xpath'
+      }
+    },
+    /**
+     * function sets up the xpath for year in expiry date of public link
+     *
+     * @param day
+     * @returns {{locateStrategy: string, selector: *}}
+     */
+    setExpiryDateDaySelectorXpath: function (day) {
+      const daySelectorXpath = util.format(this.elements.dateTimeDayPicker.selector, day)
+      return {
+        selector: daySelectorXpath,
+        locateStrategy: 'xpath'
+      }
+    },
+    /**
+     * sets provided year in expiry date field on webUI
+     *
+     * @param selector selector for expiry date "year"
+     * @returns {Promise<void>}
+     */
+    setExpiryDateYear: async function (selector) {
```

Comment:
Refactor to receive `year` as an argument instead of `selector`. Enclose that knowledge within this function.